### PR TITLE
Add Thank you Modal for survey submission

### PR DESCRIPTION
**Pull request by djnunez-aot**
*Issue #421  #435:*

-Refactored Success Panel

-Added Navigation from survey submission to engagement view page

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: met-web/src/components/survey/submit/ActionContext.tsx [hunk 1]
```diff
@@ -55,7 +55,11 @@ export const ActionProvider = ({ children }: { children: JSX.Element }) => {
                     text: 'Survey was successfully submitted',
                 }),
             );
-            navigate(`/engagement/view/${savedSurvey.engagement.id}`);
+            navigate(`/engagement/view/${savedSurvey.engagement.id}`, {
+                state: {
```
Comment (tiagograf-AOT) [inline met-web/src/components/survey/submit/ActionContext.tsx:59]: cool way of passing the state, i like it

Comment (djnunez-aot) [reply met-web/src/components/survey/submit/ActionContext.tsx:59]: thanks! :)

**Review by tiagograf-AOT — APPROVED** [review]

File: met-web/src/components/engagement/view/EngagementView.tsx [hunk 2]
```diff
@@ -27,9 +33,19 @@ export const EngagementView = () => {
         navigate(`/survey/submit/${surveyId}`);
     };
 
+    useEffect(() => {
```
Comment (jadmsaadaot) [inline met-web/src/components/engagement/view/EngagementView.tsx:36]: seems avoidable.

We can move the modal states (isEmailModalOpen and defaultPanel) to the actionContext. We can pass the setEmailModalOpen and the setDefaultPanel or handle methods that can do that using useContext() to the thankyou panel and we can on close in the success panel set the default panel as we wish

Comment (jadmsaadaot) [reply met-web/src/components/engagement/view/EngagementView.tsx:36]: You could consider a similar pattern to how the modal was handled here https://github.com/bcgov/met-public/pull/456

Comment (jadmsaadaot) [reply met-web/src/components/engagement/view/EngagementView.tsx:36]: You can also merge it, and I'll replicate that pattern

**Review by jadmsaadaot — APPROVED** [review]
Approved, just one thing to consider

we don't have to drill the isEmailModalOpen, handleClose and the defaul panel to the EmailModal we can pass it to any component using useContext like how savedEngagement is passed using useContext in the ThankYouPanel

**Pull request merged**